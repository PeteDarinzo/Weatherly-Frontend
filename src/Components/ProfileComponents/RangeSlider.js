import React, { useState, useEffect } from 'react';
import Box from '@mui/material/Box';
import Slider from '@mui/material/Slider';

function valuetext(value) {
  return `${value}°C`;
}

const farMarks = [
  {
    value: 32,
    label: '32°F',
  },
  {
    value: 45,
    label: '45°F',
  },
  {
    value: 60,
    label: '60°F',
  },
  {
    value: 80,
    label: '80°F',
  },
  {
    value: 100,
    label: '100°F',
  }
];

const celMarks = [
  {
    value: 0,
    label: '0°C',
  },
  {
    value: 10,
    label: '10°C',
  },
  {
    value: 20,
    label: '20°C',
  },
  {
    value: 30,
    label: '30°C',
  },
  {
    value: 40,
    label: '40°C',
  }
];

let minDistance;

export default function RangeSlider({ handleTempChange, units, vals }) {

  const [values, setValues] = useState([20, 40]);

  useEffect(() => {
    console.log(vals[0]);
    if (!vals[0]) {
      if (units === "imperial") {
        setValues([50, 80]);
        minDistance = 10;
      } else {
        setValues([10, 30]);
        minDistance = 5;
      }
    } else {
      setValues(vals);
    }
  }, [units])



  const handleChange2 = (event, newValue, activeThumb) => {
    if (!Array.isArray(newValue)) {
      return;
    }

    if (newValue[1] - newValue[0] < minDistance) {
      if (activeThumb === 0) {
        const clamped = Math.min(newValue[0], 100 - minDistance);
        setValues([clamped, clamped + minDistance]);
      } else {
        const clamped = Math.max(newValue[1], minDistance);
        setValues([clamped - minDistance, clamped]);
      }
    } else {
      setValues(newValue);
    }
    handleTempChange(values);
  };

  return (
    <Box my={3} px={4} py={1} pt={5} border="solid" sx={{ width: 300 }}>
      <Slider
        getAriaLabel={() => 'Minimum distance shift'}
        value={values}
        onChange={handleChange2}
        valueLabelDisplay="on"
        getAriaValueText={valuetext}
        disableSwap
        track="inverted"
        // orientation="vertical"
        min={units === "imperial" ? 32 : 0}
        max={units === "imperial" ? 100 : 40}
        marks={units === "imperial" ? farMarks : celMarks}
      />
    </Box>
  );
}