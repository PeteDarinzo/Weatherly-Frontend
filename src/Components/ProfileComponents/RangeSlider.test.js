import RangeSlider from "./RangeSlider";
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";

it("renders the range slider without crashing", () => {
  render(
    <MemoryRouter>
      <RangeSlider handleTempChange={() => { }} units="F" vals={[45, 80]} />
    </MemoryRouter>
  );
});


it("matches the range slider snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <RangeSlider handleTempChange={() => { }} units="F" vals={[45, 80]} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
