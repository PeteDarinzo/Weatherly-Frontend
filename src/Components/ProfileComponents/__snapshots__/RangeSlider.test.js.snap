// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the range slider snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kudy17"
  >
    <span
      class="MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root MuiSlider-marked MuiSlider-trackInverted css-1aznpnh-MuiSlider-root"
    >
      <span
        class="MuiSlider-rail css-1xst22u-MuiSlider-rail"
      />
      <span
        class="MuiSlider-track css-1h4p1o1-MuiSlider-track"
        style="left: 100%; width: 0%;"
      />
      <span
        class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index="0"
        style="left: 0%;"
      />
      <span
        aria-hidden="true"
        class="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
        data-index="0"
        style="left: 0%;"
      >
        0°C
      </span>
      <span
        class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index="1"
        style="left: 25%;"
      />
      <span
        aria-hidden="true"
        class="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
        data-index="1"
        style="left: 25%;"
      >
        10°C
      </span>
      <span
        class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index="2"
        style="left: 50%;"
      />
      <span
        aria-hidden="true"
        class="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
        data-index="2"
        style="left: 50%;"
      >
        20°C
      </span>
      <span
        class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index="3"
        style="left: 75%;"
      />
      <span
        aria-hidden="true"
        class="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
        data-index="3"
        style="left: 75%;"
      >
        30°C
      </span>
      <span
        class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
        data-index="4"
        style="left: 100%;"
      />
      <span
        aria-hidden="true"
        class="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
        data-index="4"
        style="left: 100%;"
      >
        40°C
      </span>
      <span
        class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
        data-index="0"
        style="left: 100%; pointer-events: none;"
      >
        <input
          aria-label="Minimum distance shift"
          aria-orientation="horizontal"
          aria-valuemax="40"
          aria-valuemin="0"
          aria-valuenow="40"
          aria-valuetext="40°C"
          data-index="0"
          max="40"
          min="0"
          step="1"
          style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
          type="range"
          value="40"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-valueLabelOpen MuiSlider-valueLabel css-1kz0hui-MuiSlider-valueLabel"
        >
          <span
            class="MuiSlider-valueLabelCircle"
          >
            <span
              class="MuiSlider-valueLabelLabel"
            >
              40
            </span>
          </span>
        </span>
      </span>
      <span
        class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb"
        data-index="1"
        style="left: 100%; pointer-events: none;"
      >
        <input
          aria-label="Minimum distance shift"
          aria-orientation="horizontal"
          aria-valuemax="40"
          aria-valuemin="0"
          aria-valuenow="40"
          aria-valuetext="40°C"
          data-index="1"
          max="40"
          min="0"
          step="1"
          style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
          type="range"
          value="40"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-valueLabelOpen MuiSlider-valueLabel css-1kz0hui-MuiSlider-valueLabel"
        >
          <span
            class="MuiSlider-valueLabelCircle"
          >
            <span
              class="MuiSlider-valueLabelLabel"
            >
              40
            </span>
          </span>
        </span>
      </span>
    </span>
  </div>
</DocumentFragment>
`;
